framework/base/ArrayableTrait.php

```diff
@@ -128,8 +154,54 @@ public function toArray(array $fields = [], array $expand = [], $recursive = tru
     }
 
     /**
+     * Extracts the root field names from nested fields.
+     * Nested fields are separated with dots (.). e.g: "item.id"
+     * The previous example would extract "item".
+     *
+     * @param array $fields The fields requested for extraction
+     * @return array root fields extracted from the given nested fields
+     */
+    public function extractRootFields(array $fields)
+    {
+        $result = [];
+
+        foreach ($fields as $field) {
+            $result[] = current(explode(".", $field, 2));
```

Comment:
Why limiting to 2?

Reply:
@sam002 I guess there is no reason for limit here since I only need the first one.

Reply:
its optimized, explode will stop after first split.